modules/tensor_mechanics/include/auxkernels/VectorRankFourAux.h

```diff
@@ -0,0 +1,38 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef VECTORRANKFOURAUX_H
+#define VECTORRANKFOURAUX_H
+
+#include "AuxKernel.h"
+#include "RankFourTensor.h"
+
+class VectorRankFourAux;
```

Comment:
The existing RankTwoAux get a value from a single RankTwoTensor. The VectorRankTwoAux gets a value from a std::vector<RankTwoTensor> variable, which come up in the viscoelastic calculation.